MoltenVK/MoltenVK/Vulkan/mvk_datatypes.mm

```diff
@@ -32,111 +32,116 @@
 
 #pragma mark Pixel formats
 
-static MVKPixelFormats _platformPixelFormats;
+static MVKPixelFormats* _platformPixelFormats = nullptr;
+
+static MVKPixelFormats* getPlatformPixelFormats() {
+	if ( !_platformPixelFormats ) { _platformPixelFormats = new MVKPixelFormats(); }
+	return _platformPixelFormats;
+}
```

Comment:
What about using a `std::unique_ptr`? Avoids minor issue when running MoltenVK via address sanitizer since I don't see the memory being freed.

Reply:
I was going to say, "there needs to be a dylib termination function to free this object," but yeah, a `std::unique_ptr` would fit the bill.

Reply:
Thanks both for the catch.

I got cheeky knowing that it will only be deleted while the library is being shut down. Although I suppose with `dylibs` that might not mean the end of proc memory. And, anyway, it's best to play nice with RAII. 

I've added the `unique_ptr`